lib/libvarnishapi/vsl_query.c

```diff
@@ -144,15 +144,40 @@ vslq_test_rec(const struct vex *vex, const struct VSLC_ptr *rec)
 
 	/* Field */
 	if (vex->lhs->field > 0) {
-		for (e = b, i = 0; *e && i < vex->lhs->field; i++) {
+		for (e = b, i = 0, dq = 0; *e && i < vex->lhs->field; i++) {
+			/* Skip end of previous field */
+			if (dq) {
+				assert(e > b);
+				assert(*e == '"');
+				dq = 0;
+				e++;
+				if (*e == '\0')
+					break;
+				if (!isspace(*e))
+					return (0);
+			}
+
 			b = e;
 			/* Skip ws */
 			while (*b && isspace(*b))
 				b++;
+
+			dq = (*b == '"');
+			if (dq)
+				b++;
 			e = b;
-			/* Skip non-ws */
-			while (*e && !isspace(*e))
-				e++;
+
+			if (dq) {
+				/* Find end of string */
+				while (*e && *e != '"')
+					e++;
+				if (*e != '"')
+					return (0);
+			} else {
```

Comment:
... and not just remove the else here, also setting `dq=0` ?

Reply:
We need to remember `dq` before we move on to the next field, so both branches really are mutual-exclusive.

See 681da75565cc897ffcb7a1921b96e17c308483ac for the two syntax errors this `if` and the other `if` blocks each catch.

Reply:
I find this confusing, especially the "field""nospace" case, tried to find a simpler solution, failed to get to one within ~20 minutes and give up for now because it's probably not relevant.
But I do wonder if we need to handle escaped quotes....